core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -398,15 +392,13 @@ memory_instantiate(WASMModuleInstance *module_inst, WASMModuleInstance *parent,
     if (heap_size > 0)
         wasm_runtime_free(memory->heap_handle);
 fail2:
-#ifndef OS_ENABLE_HW_BOUND_CHECK
-    if (memory->memory_data)
-        wasm_runtime_free(memory->memory_data);
-#else
-#ifdef BH_PLATFORM_WINDOWS
-    os_mem_decommit(mapped_mem, memory_data_size);
-#endif
-    os_munmap(mapped_mem, map_size);
+#ifdef WASM_LINEAR_MEMORY_MMAP
+    if (mapped_mem)
+        wasm_munmap_linear_memory(mapped_mem, memory_data_size, map_size);
```

Comment:
Same as above, should we restore UINT32 to 4G for memory_data_size?

Reply:
see my comment above.